### PR TITLE
Remove close button from HeaderWithCloseButton

### PR DESCRIPTION
**Pull request by staszekscp**
### Details

Removes close button from `HeaderWithCloseButton` component - expained [here](https://docs.google.com/document/d/1cQ87HcDA9RrmBM4ch7PmlNi9N9ZP8abQuPF0gKYrxDE/edit#bookmark=id.cmdh1t4935xi)

### Fixed Issues

$ https://github.com/Expensify/App/issues/15850

### Tests

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

File: src/components/HeaderWithBackButton.js [hunk 1]
```diff
@@ -87,15 +78,12 @@ const defaultProps = {
     title: '',
     subtitle: '',
     onDownloadButtonPress: () => {},
-    onCloseButtonPress: () => {},
-    onBackButtonPress: () => {},
+    onBackButtonPress: Navigation.goBack,
```
Comment (puneetlath) [inline src/components/HeaderWithBackButton.js:81]: I notice that on a lot of components we set this prop to `Navigation.goBack`, but given that is the default, why even pass this prop at all?

**Review by puneetlath — CHANGES_REQUESTED** [review]
This mostly looks good to me. Just a few things: 

1. you have lint errors 
2. I don't think we should pass the `onBackButtonPress` prop if we're just setting it to `Navigation.goBack` since that is the default 
3. Let's make sure that we do the all-device testing and include videos since this touches so many components 

cc @mountiny in case you want to review as well.

**Comment by mountiny** [comment]
Thanks @puneetlath 

Lets fix the lint issues and we can run the internal release which will help with testing. this should mainly remove the X button from all the RHP pages so there should not be a big problem hopefully

File: src/components/HeaderWithBackButton.js [hunk 2]
```diff
@@ -87,15 +78,12 @@ const defaultProps = {
...
     onThreeDotsButtonPress: () => {},
-    shouldShowBackButton: false,
```
Comment (mountiny) [inline src/components/HeaderWithBackButton.js:93]: Was there any place where the `shouldShowBackButton` was false? are we covering that case?

**Review by mountiny — COMMENTED** [review]
Thanks! Asking for a C+

File: src/components/HeaderWithBackButton.js (hunk 2 above)
Comment (mountiny) [reply src/components/HeaderWithBackButton.js:93]: There we go

File: src/components/HeaderWithBackButton.js (hunk 1 above)
Comment (staszekscp) [reply src/components/HeaderWithBackButton.js:81]: In some cases the `onBackButtonPress` should bring user back to a previous step in a flow (some screens are implemented as "steps", therefore navigation is not used)

File: src/components/ReimbursementAccountLoadingIndicator.js [hunk 3]
```diff
@@ -25,10 +24,8 @@ const propTypes = {
 
 const ReimbursementAccountLoadingIndicator = props => (
     <ScreenWrapper style={[StyleSheet.absoluteFillObject, styles.reimbursementAccountFullScreenLoading]}>
-        <HeaderWithCloseButton
+        <HeaderWithBackButton
             title={props.translate('reimbursementAccountLoadingAnimation.oneMoment')}
-            onCloseButtonPress={Navigation.dismissModal}
-            shouldShowBackButton={props.network.isOffline}
```
[the remark this answers is not in the document]
Comment (staszekscp) [reply src/components/ReimbursementAccountLoadingIndicator.js:31]: On the small screen the button is visible

[the remark this answers is not in the document]
Comment (staszekscp) [reply src/components/ReimbursementAccountLoadingIndicator.js:31]: Actually `ReportScreen` uses another component called `HeaderView` to render the back button, so this PR shouldn't affect it at all

**Comment by mountiny** [comment]
Thanks for the discussion so far, running an internal build so this is a bit easier to test https://github.com/Expensify/App/actions/runs/4811973135

**Comment by mountiny** [comment]
Tested the build on web and on my ihone and it looks great ot what I can see 
image


There is this skeleton issue with LHN on iOS for me
IMG_7944, not sure if this issue will be fixing it https://github.com/Expensify/App/pull/17962

**Review by 0xmiroslav — COMMENTED** [review]
Request money pages still have close button

Screenshot 2023-04-26 at 11 25 27 PM
Screenshot 2023-04-26 at 11 25 34 PM
Screenshot 2023-04-26 at 11 25 40 PM

File: src/components/ReimbursementAccountLoadingIndicator.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mountiny) [reply src/components/ReimbursementAccountLoadingIndicator.js:31]: You can also click the UP/Back button to close it so we are good here!

**Comment by mountiny** [comment]
> Is it expected to show back button and deprecate close button even on this kind of modal?

Oh good catch, I think the attachement modal on desktop should have X it makes more sense than <

but mobile would still have the <

**Comment by mountiny** [comment]
@0xmiroslav I would say if the attachment modal is the only place we found, I think we dont need that and we could make some custom wrapper fro that header. I think eventually we will drop usage of that modal and the gallery for pictures will be more common to what you are used to from whatsapp or messenger.

**Comment by mountiny** [comment]
I would imagine, this is not being worked on yet. But yeah, we need to keep it for now so whatever is easier now, keep the props and use it only in the attachment modal or something else. We need it there.

**Comment by staszekscp** [comment]
You may have a look now after some cleanup and fixes :)

**Review by mountiny — APPROVED** [review]
This is looking good to me 
image


I have noticed this one little bug on pronouns page and create issue for it so we dont forget https://github.com/Expensify/App/issues/18092 lets not fix it here

@0xmiroslav would you be able to go through the checklist here please? thank you!

**Review by 0xmiroslav — COMMENTED** [review]
I see you changed FlatList/index.android.js. Is this relevant to this issue?

**Review by 0xmiroslav — COMMENTED** [review]
These change original behavior. As they were added intentionally, let's make sure that there aren't any regressions after these removals.

File: src/components/HeaderWithBackButton.js [hunk 4]
```diff
@@ -63,15 +57,18 @@ const propTypes = {
...
 
     /** The guides call taskID to associate with the get assistance button, if we show it */
     guidesCallTaskID: PropTypes.string,
 
+    /** Whether we should show a close button */
+    shouldShowCloseButton: PropTypes.bool,
+
+    /** Method to trigger when pressing close button of the header */
+    onCloseButtonPress: PropTypes.func,
+
```
[the remark this answers is not in the document]
Comment (staszekscp) [reply src/components/HeaderWithBackButton.js:71]: There's an edge case in two modals, so I decided to keep it for the web

File: src/pages/iou/MoneyRequestModal.js [hunk 5]
```diff
@@ -358,18 +358,16 @@ const MoneyRequestModal = (props) => {
...
+    const navigateBack = isEditingAmountAfterConfirm ? () => navigateToStep(moneyRequestStepIndex) : navigateToPreviousStep;
     const reportID = lodashGet(props, 'route.params.reportID', '');
-    const shouldShowBackButton = currentStepIndex > 0 || isEditingAmountAfterConfirm;
     const modalHeader = (
-        <ModalHeader
+        <HeaderWithBackButton
             title={titleForStep}
-            shouldShowBackButton={shouldShowBackButton}
-            onBackButtonPress={isEditingAmountAfterConfirm ? () => navigateToStep(moneyRequestStepIndex) : navigateToPreviousStep}
+            onBackButtonPress={currentStepIndex === 0 ? Navigation.dismissModal : navigateBack}
```
[the remark this answers is not in the document]
Comment (staszekscp) [reply src/pages/iou/MoneyRequestModal.js:366]: This logic is still present, I just decided to move it to `navigateBack` instead

**Comment by staszekscp** [comment]
Hey, I've applied requested changes to [this PR](https://github.com/Expensify/App/pull/18402), because it is needed to continue work with the navigation reboot :)

**Comment by mountiny** [comment]
Closing in favour of https://github.com/Expensify/App/pull/18402

@puneetlath thanks for a review, if you are interested please feel free to review there

**Pull request closed without merging**